internal/controller/rbac/provider/roles/roles.go

```diff
@@ -53,19 +49,6 @@ var (
 	verbsSystem = []string{"get", "list", "watch", "update", "patch", "create"}
 )
 
-// Extra rules that are granted to all provider pods.
-// TODO(negz): Extra rules should be requested via a ProviderRevision's (as yet
-// unimplemented) PermissionRequests field. In order to do so we must allow the
-// RBAC manager to be configured such that it knows which rules may and may not
-// be requested.
-var rulesSystemExtra = []rbacv1.PolicyRule{
-	{
-		APIGroups: []string{""},
-		Resources: []string{pluralSecrets, pluralConfigmaps, pluralEvents},
-		Verbs:     verbsEdit,
-	},
-}
-
```

Comment:
Note that we no longer grant providers any extra rules by default; they get only the things they define. In practice pretty much all providers will need these rules. I was thinking we should update all providers packages to request them, though it occurs to me that would be a breaking change for anyone who didn't update their provider packages.

Reply:
Reverted this in a subsequent commit - I think we need to give more notice if we decide to remove these extra rules.